source/NVDAObjects/window/winword.py

```diff
@@ -1430,10 +1466,520 @@ def _get_WinwordWindowObject(self):
+                else:
+                        # Translators: Indicates that there are no series in a chart.
+                        text +=_("No Series defined.")
+                return text
+
+	def _get_title(self):
+		try:
+			title=self.wordChartObject.ChartTitle.Text
+		except COMError:
+			title=None
+		return title
+
+	def _get_role(self):
+		return controlTypes.ROLE_CHART
+
+	def getChartSegment(self):
+		chartType = self.wordChartObject.ChartType
+		if chartType in (_chartConstants.xl3DPie, _chartConstants.xl3DPieExploded, _chartConstants.xlPie, _chartConstants.xlPieExploded, _chartConstants.xlPieOfPie):
+			# Translators: A slice in a pie chart.
+			text=_("slice")
+		elif chartType in (_chartConstants.xl3DColumn, _chartConstants.xl3DColumnClustered, _chartConstants.xl3DColumnStacked, _chartConstants.xl3DColumnStacked100, _chartConstants.xlColumnClustered, _chartConstants.xlColumnStacked100, _chartConstants.xlColumnStacked):
+			# Translators: A column in a column chart.
+			text=pgettext('chart','column')
+		elif chartType in (_chartConstants.xl3DLine, _chartConstants.xlLine, _chartConstants.xlLineMarkers, _chartConstants.xlLineMarkersStacked, _chartConstants.xlLineMarkersStacked100, _chartConstants.xlLineStacked, _chartConstants.xlLineStacked100):
+			# Translators: A data point in a line chart.
+			text=_("data point")
+		else:
+			# Translators: A segment of a chart for charts which don't have a specific name for segments.
+			text=_("item")
+		return text
+
+	def invokeChartElement(self, keyIndex):
+		val=self.chartElements[self.keyList[keyIndex]]
+		func=val[0]
+		arg=val[1]
+		if arg==None:
+			func()
+		else:
+			func(arg)
+		
+	def script_nextChartElement(self,gesture):
+		if self.keyIndex == self.elementsCount-1:
+			self.keyIndex=0
+		else:
+			self.keyIndex=self.keyIndex+1
+		self.invokeChartElement(self.keyIndex)
+	script_nextChartElement.canPropagate=True
+
+	def script_previousChartElement(self,gesture):
+		if self.keyIndex == 0:
+			self.keyIndex=self.elementsCount-1
+		else:
+			self.keyIndex=self.keyIndex-1
+		self.invokeChartElement(self.keyIndex)
+	script_previousChartElement.canPropagate=True
+
+	def script_activatePosition(self,gesture):
+		# Toggle browse mode pass-through.
+		self.passThrough = True
+		self.ignoreTreeInterceptorPassThrough=False
+		browseMode.reportPassThrough(self)
+	# Translators: Input help mode message for toggle focus and browse mode command in web browsing and other situations.
+	script_activatePosition.__doc__=_("Toggles between browse mode and focus mode. When in focus mode, keys will pass straight through to the application, allowing you to interact directly with a control. When in browse mode, you can navigate the document with the cursor, quick navigation keys, etc.")
+	script_activatePosition.category=inputCore.SCRCAT_BROWSEMODE
+
+	def script_disablePassThrough(self, gesture):
+		rangeStart=self.wordApplicationObject.ActiveDocument.Range().Start

```

Comment:
Coming out of the chart, this line is making the caret move to the top of the document. Rather  a range should be passed in as a new argument to WordChart NVDAObject's constructor. Microsoft Word Inlineshape objects have a range property which can be used for this. Then, use the past in range's start property, rather than document.range().start.